src/AddressSpace.cc

```diff
@@ -881,22 +884,33 @@ void AddressSpace::unmap_internal(remote_ptr<void> addr, ssize_t num_bytes) {
 
     Mapping m = move(mm);
     mem.erase(m.map);
+
+    // Also unmap any corresponding mapping from the local address space
+    if (m.local_addr) {
+      uint8_t* start_addr =
+          m.local_addr + (max(m.map.start(), rem.start()) - m.map.start());
+      uint8_t* stop_addr =
+          m.local_addr + (min(m.map.end(), rem.end()) - m.map.end());
+      munmap(start_addr, stop_addr - start_addr);
+    }
```

Comment:
The concern I have is that, say somebody expands the mapping using remap. If we keep the local mapping, then it'll be shorter than the new remote mapping. Perhaps we should just clear the local address in an mremap?